---
title: "People"
excerpt: "People in the group"
permalink: /people/
layout: single
author_profile: false
header:
  overlay_color: "#000"
  overlay_filter: "0.3"
  overlay_image:  /assets/images/carlo-verso-378742.jpg
  caption: "Photo credit: [*Carlo Verso*](https://unsplash.com/photos/Jc-4LqyuSno)"

---


![image-left]({{ site.url }}{{ site.baseurl }}/assets/images/rivers1.png){: .align-left}
## Adam R. Rivers, PhD - PI (SY)
_PhD Massachusetts Institute of Technology / Woods Hole Oceanographic Institution, 2009_

I am a microbial ecologist turned computational biologist. I began my career woking on
nutrient limitation in marine phytoplankton and became involved in metagenomics early.
During the Deepwater Horizon oil spill I applied metatranscriptomics to
understand the fate of oil an gas released. In 2014 A became a reseach
scientist at the Joint Genome Institute, a part of Lawrence Berkeley National
Laboratory. I was the head of the Metagenome Program at JGI until 2017.  In January of 2017
I joined the Agricultural Research Service to start  the Agricultural Microbiomes Group.  

Agriculture is an exciting field to apply microbiome methods. A key question
in microbiome research is how the genetics of the host and the ecology of the
microbial community interact. Plant breeders have extensive pedigree and genotypic
information that enables us to start unraveling the interaction experimentally.
Microbiome analyses is poised to upend conventional methods for food safety
allowing us to track and eliminate pathogens, develop agricultural systems
that limit their impact through environmental manipulation. Agriculture in
an area where microbiome science will have a tremendous impact.

![image-left]({{ site.url }}{{ site.baseurl }}/assets/images/users.png){: .align-left}

## Postdoc position available

I'm just beginning to set up my group at the ARS. I have just funded a collaborative postdoc
position through Dr. Chris Reisch's lab at the University of Florida.
The position will work to develop a new method to identify in parallel all
combinations of genes that are lethal to an enteric bacteria at once. This
method will aid in the identification of antibiotic targets and help us use
 existing anti-microbials in new ways to fight infection. Because of the unique
 service-research model the position will have the opportunity to work in a high
 impact project in depth but also contribute to a number of publications that
  come through the service side of our group.

  See the position description [here](/posts/postdoc-position).

  Icon by <a href="https://www.flaticon.com/authors/smashicons" title="Smashicons">Smashicons</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a>{: .type-size-8}
