---
title: "Software"
permalink: /software/
excerpt: "Our approach to software and links to releases"
layout: single
author_profile: false
header:
  overlay_color: "#000"
  overlay_filter: "0.5"
  overlay_image: /assets/images/craylarge.jpg
  caption: "Photo credit: Adam Rivers"
---

The group is committed to publishing reproducible research by documenting the computational methods we use and making source code publicly available. We have not yet released completed software through USDA but our main account is:

[USDA-ARS-GBRU Github](https://github.com/usda-ars-gbru)

Older projects are available at:
* [arivers Github](https://github.com/arivers)
* [Berkeley Lab Bitbucket](https://bitbucket.org/berkeleylab/)

We are interested in developing containerized applications and adopting more of a devops approach to the deployment of software to the extent that it is practical on the USDA's high performance computing system, Ceres.

Software|description|Page
---------|----------------------------|------------
Reftree  | Software for querying and processing taxonomically structured data.| [link](https://bitbucket.org/berkeleylab/jgi_reftree)
iTagger | Software for processing amplicon sequencing data at a scale of 10,000 samples per year. | [link](https://bitbucket.org/berkeleylab/jgi_itagger)
Fidoplankter | An online calculator for designing iron limitation experiments in Phytoplankton, written before I knew about Github :) |[link](http://fidoplankter.uga.edu/)
ARS Microbiome workshop | Website for a training held in August 2017 | [link](https://usda-ars-gbru.github.io/Microbiome-workshop/)
ARS-RQC | Still in development, an an automated Illumina quality control workflow and visualization web application | [link](https://github.com/USDA-ARS-GBRU/ARS-RQC)
